### DISCUSSION TITLE
Re-render cookie consent after cookieconsent.run(<config_object>) has already been executed

### DISCUSSION THREAD

**Discussion by Chayemor** [orestbida/cookieconsent#489] [Q&A] [answered]
I have been tasked with creating a "Read more" link inside of the text that appears on the `<config_object>` 's` ["consent_modal"]["description"]` part. 

I have that working, the link appears and it changes the css classes on the onClick event, but I see that the modal is not rerendering with the new `config_object `. 

Is there a way I can force a re-render of the dialogue so that the new `description` now shows?
---

I have tried using the "revision" number, but it's not working. 

```javascript
  useEffect(() => {
    console.log("running again with ", displayMoreText);
    console.log("revision number is  ", readMore ? 1 : 0);

    const cc = window.initCookieConsent();

    window["_vanillaCookieConsentUtils"] = {
      "setReadMore": setReadMore
    };

    cc.run({
      revision: readMore ? 1 : 0,
      page_scripts: true,
      gui_options: guiOptions,
      languages: getLanguages(displayMoreText),
     ....
    });
  }, [readMore]);
```
-----

Thank you!

**Comment by orestbida** [comment, marked as answer]
The plugin is not a react component so any state change won't be reflected unless you completely reset and then re-run the plugin.

If possible only call  `run()` once, as shown in this [demo](https://stackblitz.com/edit/nextjs-euxk9k?file=components%2FCookieConsentComponent.js).

You can use `updateLanguage('<lang>', true)` to refresh the current translation in both modals.

E.g.

```javascript
cc.getConfig('languages').en.consent_modal.title = 'New title';
cc.updateLanguage('en', true);
```

Note: [v3](https://github.com/orestbida/cookieconsent/tree/v3.0-beta#cookieconsent-v3) is more suited for react based frameworks

**Reply by Chayemor** [reply]
That was it, thank you! 

I'll try updating to 3 soon, thank you!
